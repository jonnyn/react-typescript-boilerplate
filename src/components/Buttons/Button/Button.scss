.btn-icon {
  all: unset;
  font-size: 22px;
  background-color: blue;
  color: white;
  transition-timing-function: ease-in-out;
  transition-duration: 300ms;
}

.btn-icon:hover {
  color: white;
  opacity: 0.5;
  transition-timing-function: ease-in-out;
  transition-duration: 300ms;
}

.btn-icon:active {
  opacity: 1;
  transition-timing-function: ease-in-out;
  transition-duration: 300ms;
}
