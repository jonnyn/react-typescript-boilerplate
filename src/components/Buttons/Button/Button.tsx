import './Button.scss'

type Props = {
  label?: string
  className?: string
  style?: object
  children?: JSX.Element
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
  disabled?: boolean
}

const Button = (props: Props) => {
  const { label, className, style, onClick, children, disabled } = props
  return (
    <button
      type="button"
      data-testid="button"
      style={style}
      onClick={onClick}
      className={className || 'btn-icon'}
      disabled={disabled}
    >
      {label}
      {children}
    </button>
  )
}

export default Button
