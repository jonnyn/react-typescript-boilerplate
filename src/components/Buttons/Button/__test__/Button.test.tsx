import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Button from '../Button'

afterEach(cleanup)

// eslint-disable-next-line jest/expect-expect
it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Button />, div)
})

it('renders button correctly with label', () => {
  const { getByTestId } = render(<Button label="click test" />)
  expect(getByTestId('button')).toHaveTextContent('click test')
})

it('renders button correctly with disabled', () => {
  const { getByTestId } = render(<Button disabled />)
  expect(getByTestId('button')).toHaveAttribute('disabled')
})

it('renders button correctly with children', () => {
  const { getByTestId } = render(
    <Button>
      <div>Back</div>
    </Button>,
  )
  expect(getByTestId('button')).toHaveTextContent('Back')
})

it('matches snapshot', () => {
  const tree = renderer.create(<Button label="Back" />).toJSON()
  expect(tree).toMatchSnapshot()
})
