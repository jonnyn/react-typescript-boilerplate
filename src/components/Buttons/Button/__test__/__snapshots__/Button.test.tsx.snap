// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="btn-icon"
  data-testid="button"
  type="button"
>
  Back
</button>
`;
